### ISSUE TITLE
Reland #957

### ISSUE THREAD

**Issue by rylev** [rust-lang/rustc-perf#961]
#960 is a revert of #957. We did this revert because we were seeing errors in production after it merged. We need to investigate the errors and reland #957. Unfortunately, the errors are likely hidden in the logs which only @Mark-Simulacrum has access to.

**Comment by Mark-Simulacrum** [comment]
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E42601), message: "subquery in FROM must have an alias", detail: None, hint: Some("For example, FROM (SELECT ...) [AS] foo."), position: Some(Original(78)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("gram.y"), line: Some(11895), routine: Some("base_yyparse") }) }', database/src/pool/postgres.rs:368:30
```

**Comment by tgnottingham** [comment]
Thanks to both of you for handling the fallout. :) I'll be sure to test on all DB types in the future. Should be fixed in #964.
